services/registration/src/routes/application.ts

```diff
@@ -10,16 +11,40 @@ export const applicationRouter = express.Router();
 applicationRouter.route("/").get(
   asyncHandler(async (req, res) => {
     const applications = await ApplicationModel.find({});
+    const userIdArray: string[] = [];
+    for (const application of applications) {
+      userIdArray.push(application.userId);
+    }
+
+    const userInfo = apiCall(
+      Service.USERS,
+      { method: "POST", url: `users/actions/retrieve`, data: userIdArray },
+      req
+    );
+
+    const finalApps = {
+      ...applications,
+      users: userInfo,
+    };
```

Comment:
For this, can you add the user info onto the application, and now have it separate? So you'll have to loop through all the userInfo that's returned, and match the`userId` field to the `userId` field of each application, and then add all the data like `user: ----`.